Lib/typing.py

```diff
@@ -524,11 +524,10 @@ def _evaluate(self, globalns, localns):
     def __eq__(self, other):
         if not isinstance(other, ForwardRef):
             return NotImplemented
-        return (self.__forward_arg__ == other.__forward_arg__ and
-                self.__forward_value__ == other.__forward_value__)
+        return self.__forward_arg__ == other.__forward_arg__
```

Comment:
So it seems best to keep the check for forward value, contingent upon them both existing:

```suggestion
        if self.__forward_evaluated__ and other.__forward_evaluated__:
            return (self.__forward_arg__ == other.__forward_arg__ and
                    self.__forward_value__ == other.__forward_value__)
        return self.__forward_arg__ == other.__forward_arg__
```

I think that this, combined with the new `__hash__`, is probably the most useful and least surprising implementation.